### PR TITLE
Fix batch processing of log tables retaining previous entries

### PR DESCRIPTION
**Pull request by edmofro**
Juliana has recently been noticing some  issues with bulk imports - sometimes not all of the data will make it in. I believe this is the reason

### CODE REVIEW

File: BuildScripts/mvComplexFunctions.sql [hunk 1]
```diff
@@ -917,6 +917,7 @@ BEGIN
 
             tLastType               := tDmlType;
             iArraySeq               := 1;
+            uRowIDArray             := '{}';
```
Comment (edmofro) [inline BuildScripts/mvComplexFunctions.sql:920]: This section of code is batching the processing of log table entries into groups with the same change type. So if there are 8 `UPDATE` changes to answers, then 8 `DELETE` changes to other answers, it'll do the updates all in one batch, then the deletes all in one batch. 

The problem was when you get e.g. 8 `UPDATE` changes followed by say 4 `DELETE` changes, it'll batch the 8 updates, process them, and then batch the 4 deletes but _use the same array as before_, so the indexes that haven't been overwritten still have the last 4 updates in them. That results in those 4 getting deleted from the analytics table 🤦

**Comment by edmofro** [comment]
Also interested in whether you think we should submit this back to the original repo as a PR @rohan-bes?

**Comment by rohan-bes** [comment]
Wow! Incredible investigation on this one @edmofro ! Yes, I do think we ought to put this PR back to the mainline, seems like a genuine bugfix!

**Review by rohan-bes — APPROVED** [review]
Looks good to me!

**Comment by edmofro** [comment]
Thanks @rohan-bes! Credit also goes to @EMcQ-BES who paired with me on this

Have created a PR back to the base repo https://github.com/CDLSoftware/pg-mv-fast-refresh/pull/45

**Pull request merged**